src/realm/object-store/c_api/conversion.hpp

```diff
@@ -25,14 +25,17 @@ static inline realm_string_t to_capi(StringData data)
     return realm_string_t{data.data(), data.size()};
 }
 
-static inline realm_string_t to_capi(const std::string& str)
+static inline realm_string_t to_capi(util::Optional<std::string_view> str)
```

Comment:
```suggestion
static inline realm_string_t to_capi(const util::Optional<std::string_view>& str)
```
Just a question: is it enough that the function is `inline` to make sure we don't copy `util::Optional` here, or do we also weed to pass it as a const reference?